fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RewriteMultiDistinctByCTERule.java

```diff
@@ -133,17 +133,18 @@ public RewriteMultiDistinctByCTERule() {
     public boolean check(OptExpression input, OptimizerContext context) {
         // check cte is disabled or hasNoGroup false
         LogicalAggregationOperator agg = (LogicalAggregationOperator) input.getOp();
-        if (!context.getSessionVariable().isCboCteReuse()) {
-            return false;
-        }
-
-
         List<CallOperator> distinctAggOperatorList = agg.getAggregations().values().stream()
                 .filter(CallOperator::isDistinct).collect(Collectors.toList());
         boolean hasMultiColumns = distinctAggOperatorList.stream().anyMatch(f -> f.getChildren().size() > 1);
+        if (hasMultiColumns && distinctAggOperatorList.size() > 1) {
+            return true;
```

Comment:
For multiple distinct cols, we will transform it to CTE no matter with the cbo_cte_reuse value?

Reply:
yes